### DISCUSSION TITLE
Converting Metes and Bounds Surveys to Lat-Long and Shapefiles

### DISCUSSION THREAD

**Discussion by taylorterry3** [geopandas/geopandas#2801] [Ideas]
I recently had to hand-roll some Python code to convert a metes and bounds legal description to a shapefile. I did a fair bit of looking for open source code before doing it myself and didn't find anything, so before I file this stuff away I had the thought that I might be able to add it to some open source project. I'm not sure Geopandas is necessarily the right place for this, but I figure this community might be able to give some guidance as to where this kind of thing might belong (or tell me to just package it as a standalone tool).

For context, metes and bounds is the system used in much of the US to record the legal description of a property for a deed. It's basically a list of directions and distances describing the property boundary starting from some known point. Turning this into a series of lat-long points is at the core just a simple exercise in trigonometry for which GeoDataFrames are well suited, but there is a ton of lurking complexity due to everything from magnetic vs. true north directions and changes in magnetic declination over time, to survey error causing the shapefile to not close without adding an error segment.

Any guidance or pointers are appreciated!

Thanks,
Taylor

**Comment by martinfleis** [comment]
Thanks for reaching out. This is surely not a feature that belongs to geopandas. The closest package to something like this I can think of is probably @walkerke's [pygris](https://github.com/walkerke/pygris). He's surely more aware of the US context than I am so he may either be interested in including this or point you to a right direction.

**Reply by taylorterry3** [reply]
Thanks for the pointer! Pygris itself is probably not quite the right place as it deals only with the US government's own geometries AFAIK, but I'll wait for @walkerke weigh in.

**Comment by walkerke** [comment]
Thanks for the ping!  @taylorterry3 you are correct, pygris only deals with the US Census Bureau's geometries and doesn't handle metes / bounds.

**Reply by taylorterry3** [reply]
Much obliged @walkerke! Any suggestions for some other toolkit that this might belong within, or should I just package it up independently?

**Comment by Nexicobob** [comment]
I'm new to Github and am limited in my programming knowledge but I have a client that would be interested in converting metes and bounds text to LatLng coordinates. What is a proper way for us to connect? Thanks

**Reply by martinfleis** [reply]
If your client would be interested in funding the development of the conversion feature that could be implemented in geopandas or other package, drop me an email at martin@martinfleischmann.net. We can discuss further via email.

**Reply by taylorterry3** [reply]
I'm the originator of this issue. I never got around to packaging this code, but if there is renewed interest please let me know.

**Comment by Nexicobob** [comment]
Taylor,
Converting metes and bounds text, because they come in various forms, looks
to be very tricky.
If it can be done well, my client is interested.

Will your package be able to accept raw metes and bounds text from a legal
description or will "massaging" or preprocessing be needed?

I am just starting the conversation to see if we can find a match.
Thanks,
Bob

Bob Tablak
(408) 313-2814

On Sat, Feb 10, 2024 at 1:18 PM Taylor Terry ***@***.***>
wrote:

> I'm the originator of this issue. I never got around to packaging this
> code, but if there is renewed interest please let me know.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/geopandas/geopandas/discussions/2801#discussioncomment-8429743>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BGCDA4CKAU2XSFWEVVSCVLLYS7PZ3AVCNFSM6AAAAAAVH675FCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DIMRZG42DG>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Bob Tablak
408 313-2814

**Reply by taylorterry3** [reply]
This is probably better taken offline, but LLMs do a better job with metes and bounds than any bespoke parser ever could. Someone only doing a couple of properties would probably just want to go that route or even just hand enter it if there weren't that many corners. To do this at scale you'd need a custom solution anyway, as you need some data that isn't directly in the text of the legal description. The most important thing is the lat-long of the starting point, but you also need to know if the survey is old enough to have used magnetic rather than true north and then figure out what the declination was for that geography in that year.

If you want to discuss further my email address is my github username without the number three at the end at gmail.

Anyone else reading this, I am still interested in open-sourcing at least the basics of this process.

**Comment by mrdvt92** [comment]
> convert a metes and bounds legal description to a shapefile

I wrote a [Perl CGI script](https://github.com/mrdvt92/perl-Data-Metes_and_Bounds) in 2001 and since it was written in Perl it still runs fine. [example](http://www.landsalesinc.com/cgi-bin/deedcallparse.cgi)

If you know the coordinates of the start, you could also use the NGS program in the public domain "forward" by Robert (Sid) Safford and Stephen J. Frakes ([Perl port](https://metacpan.org/dist/Geo-Forward)) to calculate the coordinates of subsequent points.  You would have to figure out how to convert magnetic north to WGS-84 or something similar like NAD83.
